core/src/test/java/org/elasticsearch/common/settings/SettingsTests.java

```diff
@@ -68,6 +68,17 @@ public void testReplacePropertiesPlaceholderSystemProperty() {
         assertThat(settings.get("setting1"), equalTo(value));
     }
 
+    public void testReplacePropertiesPlaceholderSystemPropertyList() {
+        String value = System.getProperty("java.home");
+        assertFalse(value.isEmpty());
+        Settings settings = Settings.builder()
+            .put("property.placeholder", value)
+            .putList("setting1", "${property.placeholder}", "${property.placeholder}")
+            .replacePropertyPlaceholders()
```

Comment:
We have `replacePropertyPlaceholders(Function<String, String>)`; we use this rather than pulling from the same settings object as I think this will be a more realistic test?